drift/core/beamtransfer.py

```diff
@@ -675,30 +515,29 @@ def _generate_mfiles(self, regen=False):
 
         # Get the frequencies and baselines that we are going to calculate and create a
         # lookup table for the order in which we are going to calculate them
-        freq_to_include = [
-            fi
-            for fi in range(self.telescope.nfreq)
-            if not self.telescope._skip_freq(fi)
-        ]
-        baselines_to_include = [
-            bi
-            for bi in range(self.telescope.nbase)
-            if not self.telescope._skip_baseline(bi)
-        ]
-        nfb = len(freq_to_include) * len(baselines_to_include)
+        freq_to_include = self.telescope.included_freq
+        baselines_to_include = self.telescope.included_baseline
+
+        # Calculate the number of included entries
+        nf_inc = len(self.telescope.included_freq)
+        nb_inc = len(self.telescope.included_baseline)
+        np_inc = len(self.telescope.included_pol)
+        nl = self.telescope.lmax + 1
+        nm = self.telescope.mmax + 1
+
+        nfb = nf_inc * nb_inc
 
         fbmap = np.array(
             np.meshgrid(freq_to_include, baselines_to_include, indexing="ij")
         ).reshape(2, nfb)
 
+        fbcompact = np.array(
+            np.meshgrid(np.arange(nf_inc), np.arange(nb_inc), indexing="ij")
+        ).reshape(2, nfb)
+
         # Calculate the number of baselines to deal with at any one time. Aim
         # to have a maximum of "mem_chunk" GB in memory at any one time
-        fbsize = (
-            self.telescope.num_pol_sky
-            * (self.telescope.lmax + 1)
-            * (2 * self.telescope.mmax + 1)
-            * 16.0
-        )
+        fbsize = self.telescope.num_pol_sky * nl * 2 * nm * 16.0
```

Comment:
Should it be `np_inc` instead of `self.telescope.num_pol_sky` here?

Reply:
I *think* it actually still needs to be num_pol_sky for the moment anyway. The reason is that some of the internal steps still allocate the memory for the full num_pol_sky when generating the BTMs, and then it's only trimmed back to np_inc as it is about to write out to disk.